pkg/core/tekton/tekton.go

```diff
@@ -454,3 +498,88 @@ func getInputCodesetPath(inputs []*workflow.WorkflowStepInput) string {
 	}
 	return ""
 }
+
+func (w *WorkflowBackend) toWorkflowRun(wf workflow.Workflow, p v1beta1.PipelineRun) *workflow.WorkflowRun {
+	startTime := p.Status.StartTime.Format(time.RFC3339)
+	wfr := workflow.WorkflowRun{
+		Name:        &p.ObjectMeta.Name,
+		WorkflowRef: &wf.Name,
+		StartTime:   &startTime,
+	}
+
+	if p.Status.CompletionTime != nil {
+		completionTime := p.Status.CompletionTime.Format(time.RFC3339)
+		wfr.CompletionTime = &completionTime
+	}
+
+	for _, input := range wf.Inputs {
+		var value string
+		if *input.Type == inputTypeCodeset {
+			value = fmt.Sprintf("%s:%s", *getPipelineResourceParamsValue("url", p.Spec.Resources[0]),
+				*getPipelineResourceParamsValue("revision", p.Spec.Resources[0]))
+		} else {
+			value = *getPipelineRunParamValue(*input.Name, p.Spec.Params)
+		}
+		wfr.Inputs = append(wfr.Inputs, &workflow.WorkflowRunInput{
+			Input: input,
+			Value: &value,
+		})
+	}
+
+	for _, output := range wf.Outputs {
+		wfr.Outputs = append(wfr.Outputs, &workflow.WorkflowRunOutput{
+			Output: output,
+			Value:  getPipelineRunResultValue(*output.Name, p.Status.PipelineResults),
+		})
+	}
+	status := "Unknown"
+	if len(p.Status.Conditions) > 0 {
+		status = pipelineReasonToWorkflowStatus(p.Status.Conditions[0].Reason)
+	}
+	wfr.Status = &status
+
+	url := fmt.Sprintf("%s/#/namespaces/%s/pipelineruns/%s", w.dashboardURL, w.namespace, *wfr.Name)
+	wfr.URL = &url
+	return &wfr
+}
+
+func pipelineReasonToWorkflowStatus(reason string) string {
+	return strings.TrimPrefix(reason, "PipelineRun")
+}
+
+func getPipelineResourceParamsValue(paramName string, resource v1beta1.PipelineResourceBinding) *string {
+	for _, param := range resource.ResourceSpec.Params {
+		if param.Name == paramName {
+			return &param.Value
+		}
+	}
+	return nil
+}
+
+func getPipelineRunParamValue(paramName string, params []v1beta1.Param) *string {
+	for _, p := range params {
+		if p.Name == paramName {
+			return &p.Value.StringVal
+		}
+	}
+	return nil
+}
+
+func getPipelineRunResultValue(resultName string, results []v1beta1.PipelineRunResult) *string {
+	for _, p := range results {
+		if p.Name == resultName {
+			return &p.Value
+		}
+	}
+	return nil
+}
+
+func contains(slice []string, item string) bool {
+	set := make(map[string]struct{}, len(slice))
+	for _, s := range slice {
+		set[s] = struct{}{}
+	}
+
+	_, ok := set[item]
+	return ok
+}
```

Comment:
This function creates a map out of a slice just to be able to use it's hashing capabilities to do a search, am I right ? Is this really the optimal golang way to search a string in a slice, in case I may need it in the future ? thanks !

Reply:
You know, on second thought, in this context it's probably overkill to do this just to achieve o(1) search time, given that your slice is probably only going to be a couple of items in size max. I'd just iterate the slice in your place.

Reply:
That might be true for today but chances are high that I will need to use such function in the future, so I prefer to keep it as it is.

Reply:
Okay, but you're adding unnecessary complexity, which I suggest you avoid if possible.

Reply:
I'm not gonna tell you to remove it, but could you explain what's the point?

Even if _chances are high that I will need to use such function in the future_, how is it more efficient than just going over the slice? You need to do it anyway when creating that map, and that's true for large slices as well as small one. Unless you are actually saving that map of indexes somewhere, there's no added value here.

Reply:
You are right the way it is does not add any value.